### ISSUE TITLE
Keras model functional build of a dynamic lambda layer fails with `RecursionError`

### ISSUE THREAD

**Issue by nicolaspi** [tensorflow/tensorflow#47879]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.4.1
- Python version: 3.8

**Describe the current behavior**
Building a model with dynamic lambda layer fails with exception `RecursionError`

**Describe the expected behavior**
The model should build without error.

**Standalone code to reproduce the issue**

```python
import tensorflow as tf

input = tf.keras.Input(shape=())
output = tf.keras.layers.Lambda(lambda x: x, dynamic=True)(input)
tf.keras.Model(inputs=input, outputs=output)
```
Output:
`RecursionError: maximum recursion depth exceeded while calling a Python object
`

**Comment by amahendrakar** [comment]
@nicolaspi,
This issue has already been addressed in issue [#44906](https://github.com/tensorflow/tensorflow/issues/44906) by members of the TensorFlow community. Could you please go through it. Thanks!

**Comment by nicolaspi** [comment]
Indeed, this is a duplicate.
